### PR TITLE
chore: fix warnings, improve code style as preparation for the main PR of [WPB-4549]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4549" title="WPB-4549" target="_blank">StoryWPB-4549</a>  [iOS] Rely on remote search when searching for users
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

A few extracted and easy to review changes from PR https://github.com/wireapp/wire-ios/pull/870.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import Foundation
 import WireDataModel
 
-final class ConversationEventPayloadProcessor {
+struct ConversationEventPayloadProcessor {
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Payloads/Processing/ConversationEventPayloadProcessor.swift:22]: 🎉

File: wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift [hunk 2]
```diff
@@ -27,7 +28,7 @@ protocol MLSEventProcessing {
 
 }
 
-final class MLSEventProcessor: MLSEventProcessing {
+struct MLSEventProcessor: MLSEventProcessing {
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift:31]: 🎉

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift [hunk 3]
```diff
@@ -713,7 +719,7 @@ class ConversationByQualifiedIDTranscoder: IdentifierObjectSyncTranscoder {
     }
 }
 
-class ConversationByIDListTranscoder: IdentifierObjectSyncTranscoder {
+final class ConversationByIDListTranscoder: IdentifierObjectSyncTranscoder {
```
Comment (markusfassbender) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift:722]: 👍

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift [hunk 4]
```diff
@@ -140,8 +141,11 @@ public class ConversationRequestStrategy: AbstractRequestStrategy, ZMRequestGene
             updateAccessRolesActionHandler,
             updateRoleActionHandler,
             SyncConversationActionHandler(context: managedObjectContext),
-            CreateGroupConversationActionHandler(context: managedObjectContext, mlsService: mlsService),
-            UpdateConversationProtocolActionHandler(context: managedObjectContext)
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationRequestStrategy.swift:144]: just notice this was removed, was it intentional ?

**Pull request merged**